List Resources (prototype UI)

jameshadfield opened the pull request:
This (draft) PR represents a bunch of ideas about how we can present datasets. At the moment it is working to display all of the core datasets - check it out via [the /pathogens URL on the review app](https://nextstrain-s-james-list-pbsqum.herokuapp.com/pathogens). I wanted to convey:

* The breadth of datasets (e.g. there are 330 different datasets under `/ncov/*`). This includes the ability to filter to datasets by name (or parts of name, e.g. "ha" or "h3n2")
* The depth of datasets (e.g. for those 330 ncov datasets there are over 44,000 snapshots available) - there are 3 different viz ideas I'm using to display them, but I expect I'll converge on 1 or 2 in the end.
* The ability to load previous snapshots. Currently these links are revealed by shift-clicking on the dataset name (or click on the version icon). Please et me know if this wasn't obvious.
* Still keep the idea of "showcase" datasets or "quick link" datasets so that it's not too daunting!

Eventually I hope that this UI will expand in usage to other resources (narratives, intermediate files) as well as other resource sources (staging, groups, perhaps on the main splash page in some form?) 

Please try it out and send comments -- what worked well, what didn't, and any ideas you may have for improvements to the UI. Thanks!

### Code quality, styling tweaks, little bugs

At the moment it's a series of ideas. I've iterated on the ideas quite a lot, so I think the _ideas_ are good, but I haven't spent time cleaning up the styling or code for each idea as I wanted to discuss the ideas before moving on to that stage.

huddlej reviewed (commented):
This is really cool, @jameshadfield! I wrote up some quick thoughts based on my initial interactions with the new interface.

* I like a lot about this prototype! Some of my favorite bits are:
  * the showcase links at the top
  * the type-to-filter and how incredibly fast it is
  * the design of each pathogen’s card where each level of the hierarchy is only shown the first time, so you can clearly see the nesting of datasets
  * the details on demand when hovering over a dataset link
  * the modal window you get when you click to see past datasets (I didn’t know what to expect when I clicked on the little “refresh” icon, but I was pleasantly surprised by the display and how playful the dot plot appears at first glance) and I like that clicking a dot opens the snapshot in a new window
  * the “quick links” at the top of some cards as a way to highlight the best entry points to different pathogens
* I wish that when I hovered over the sparkline for each dataset that I saw the same details on demand that I get when I hover over the dataset link.
* For the modal window of past snapshots, I wish that the dots were bigger (easier to hover over and click, although I see now that the size varies by dataset and maybe by number of points in each column) and that there was a y-axis that visually indicated the order of the dots in each column. It also looks like dots aren’t grouped into columns by month since the same column can have snapshots from October, November, and December. What if they were actually grouped by month?
  * Also, what if the modal window had a standard “x” icon in a top corner along with a “close” link so users don’t have to read text about how to close the window? The click-outside-the-window pattern seems common enough these days that folks who are comfortable with that approach will already try it without guidance.
  * Also also, you probably don’t have to literally type “fine print” 😆
* I wish the icons for “total number of datasets” and “total number of versions” were a little more clearly mapped to those concepts. The current icons convey to me concepts/actions of “share this” and “refresh this”, respectively. That might just be me, though.
* I wish that each card’s header specified the human-readable name of the dataset like “SARS-CoV-2 datasets last updated 2024-02-28” instead of “Datasets last updated 2024-02-28”.
* Some of the cards include datasets for multiple distinct pathogens or appear incomplete. For instance, there is a card with a single mpox lineage B.1 dataset followed by another card with two more mpox datasets and the seasonal flu B/Vic dataset (below). Similarly, Zika is grouped with subsets of Vic, H3, and H1 datasets.

<img width="1116" alt="image" src="https://github.com/nextstrain/nextstrain.org/assets/85372/85dd61bd-ab4c-4ec0-b66d-56f8a2e33f22">

* When I click on one showcase group, the data get filtered to that group, but then when I click on a different showcase group, nothing happens. I wasn’t sure what I expected to have happen but maybe for the new showcase group’s data to replace the previously selected group.

jameshadfield commented:
Thanks @huddlej - great comments. I'll wait for others' before doing any more work here, but I think yours are generally both sensible and actionable. 

> I wish that each card’s header specified the human-readable name of the dataset like “SARS-CoV-2 datasets last updated 2024-02-28” instead of “Datasets last updated 2024-02-28”...  Some of the cards include datasets for multiple distinct pathogens or appear incomplete. For instance, there is a card with a single mpox lineage B.1 dataset followed by another card with two more mpox datasets and the seasonal flu B/Vic dataset (below). Similarly, Zika is grouped with subsets of Vic, H3, and H1 datasets.

There are two different grouping approaches here. The default, which is used in your screenshot, groups datasets of _any_ pathogen by the last-updated date. So there can be a mix of some mpox, flu etc within the same group. If you toggle to "alphabetical" you'll get the grouping by pathogen. So either the toggle wasn't obvious, or what it did wasn't obvious, and/or the default was wrong. Perhaps `/alphabetical/by pathogen/`?

<img width="472" alt="image" src="https://github.com/nextstrain/nextstrain.org/assets/8350992/c843ce5b-b029-49c9-aae8-05d11c7bbab8">

huddlej commented:
Ah, thanks, @jameshadfield! I did miss that grouping option. I expected the default to be grouping by dataset (since I'm a user who cares most about flu and would like to see all those data together) and then ordering of those datasets by most recently updated and an option to order alphabetically. To get this display for flu, it seems I should select the showcase tile for "seasonal flu" or type "flu" and "seasonal" into the filter display.

huddlej commented:
After the live demo today, my only maybe blocking request would be for the default grouping to be by "pathogen (sorted alphabetically)" instead of the most recently updated, but otherwise this seems immediately useful even if we refine the aesthetics more later...

trvrb commented:
This is fantastic work @jameshadfield! We just talked through most of my review, but I'll briefly summarize main suggestions here:
1. Keep one pathogen per card and sort cards alphabetically by pathogen name or by last updated date
2. Quicker _main link_. Ie clicking on card title of "rsv" or "ncov" should be the equivalent of going to `/rsv` or `/ncov`. 
3. The modal box with individual snapshots doesn’t flex well between smaller collections and larger collections. There should be a simple way to dial in a desired particular date rather than needing to identify a corresponding snapshot. I imagine this is actually the main way that someone would want to interact here. I could imagine a slider through time along the x-axis in the current beeswarm plot and feedback as snapshot changes. Ie UI to select a date, rather than UI to select a particular snapshot.
4. Sorting datasets within a pathogen card based on [`manifest_core.json`](https://github.com/nextstrain/nextstrain.org/blob/master/data/manifest_core.json). This would be the natural way to order `ncov_gisaid_global`, `ncov_gisaid_africa`, etc... or `seasonal-flu/h3n2`, `seasonal-flu/h1n1pdm`, etc...

Small critiques:
1. I'd suggest replacing "Last updated between 2020-02-21 & 2024-03-05" with "Last updated 2024-03-05", just to be less verbose.
2. Modify the icons for dataset count and snapshot count. For me these are semantically "share" and "refresh".
3. Collapse carat is hard to restore due to it jumping down the page. Suggest moving the carat to the top of the card.
4. It would be nice if the top of each of the three columns should have the full path spelled out.
5. Semantically, I’d expect the sparkline to live next to the updated counter.

One thing that we didn't talk about: I do think it's pretty essentially that the page is responsive enough to shrink down to phone size width. Currently it looks like:

![Screenshot 2024-03-07 at 3 27 21 PM](https://github.com/nextstrain/nextstrain.org/assets/1176109/566e7572-19be-4991-ace1-a485c552607e)

Given the complexity of the UI it's going to be difficult to keep all the functionality on a phone display. My suggestion here would be once the display width is below a particular value that you hide the sparkline and hide the snapshot counter, so effectively just creating a long list of dataset names to tap on. Ie don't attempt to surface the snapshot select modal and instead just have it get to the current dataset.

trvrb commented on src/resourceIndex.js:
```diff
@@ -128,8 +128,51 @@ async function updateResourceVersions() {
+    /**
+     * TODO - don't hardcode these, use our routing rules instead. TODO XXX
+     */
+    const skipFirstWords = [
```
I just deleted the following from the `nextstrain-data` bucket:
- `zika_test.json`
- `zika-tutorial-metadata-via-api.json`
- `zika-v2`

I couldn't find any JSONs in `nextstrain-data` bucket that began with:
- `ncovRegional`
- `ncov-country`
- `ncov-priorities`
- `ncov-with-priorities`
- `ncovRefocus`
- `ncov-test`
- `seattle`

jameshadfield replied:
> I couldn't find any JSONs in `nextstrain-data` bucket that began with: `ncovRegional` [...]

If you toggle "Show versions" we can see what's going on here:

<img width="504" alt="image" src="https://github.com/nextstrain/nextstrain.org/assets/8350992/f40ca1b8-b718-4188-919c-7fac853597db">

With [how I handle delete markers](https://docs.nextstrain.org/en/latest/guides/snapshots.html#s3-delete-markers) I'm still exposing all the (~600) versions prior to the one that the delete marker "deleted". These datasets will be hidden from the UI once I import the routing rules into the API which hands us resource-index data, so it won't be an issue for the UI. But if we really want to clean up the bucket we'll have to consider old versions as well. (I don't have strong feelings on whether this is worth it.)

trvrb replied:
Got it. Thank you! This was my same confusion on Slack with the https://nextstrain.org/ncov/north-america/fft@2021-01-13. I do think the indexer is doing the correct thing here. I've just gone through and deleted past versions of:
- `zika-test`
- `zika-tutorial-metadata-via-api`
- `zika-v2`
- `ncovRegional`
- `ncov-country`
- `ncov-priorities`
- `ncov-with-priorities`
- `ncovRefocus`
- `ncov-test`
- `seattle`

joverlee521 commented:
We chatted about this as a group in today's project meeting, but I wanted to jot down my feedback here. I really like the pathogen grouped display, I think it is much more usable than the current page! The dataset modals that exposes the past snapshots are very helpful! No more trying random dates in the URL
     
Some aesthetic nitpicks:
- The word "updated" can be misleading because here it means when the dataset was uploaded to S3, maybe "published" is more appropriate? Or just rely on the word "snapshots", i.e. "Snapshots between 2023-12-13 & 2024-02-06"
- I was initially confused by the empty pipes, I thought there was some UI issue that prevented the words from loading. I wonder if a [tree view](https://mui.com/x/react-tree-view/#controlled-tree-view) of the datasets would help here?
- The per dataset sparklines take up a lot of room on the page. Maybe they can be summarized as a single sparkline per pathogen and only display the per dataset sparklines on hover? 
- Echoing the icons are misleading for datasets/snapshots. I'd suggest a [bullet list](https://react-icons.github.io/react-icons/search/#q=bullet%20list) for datasets and [albums](https://react-icons.github.io/react-icons/search/#q=albums) for snapshots.
     
Questions:
- Thoughts on separating grouping and sorting options? I'd like to group by pathogens but sort by published date, where the top entry is the pathogen with the more recent dataset.
- Do you plan to add links to the individual pathogen pages? (e.g. [https://nextstrain.org/sars-cov-2/](https://next.nextstrain.org/sars-cov-2/))

trvrb commented:
>Do you plan to add links to the individual pathogen pages? (e.g. [https://nextstrain.org/sars-cov-2/](https://next.nextstrain.org/sars-cov-2/))
I see these "dataset collection" pages for `/sars-cov-2` and `/influenza` as separate from the dataset UI. I'd think maybe in this case links to these two pages should live above the dataset UI. However, I see it's going to be quite unclear that these tiles go directly to a different page on nextstrain.org vs the lower tiles that are part of the dataset UI act as filters.

I could see a different strategy where we keep a selection of hand-curated "showcase" tiles above the dataset UI that are direct links. This would work similarly to the existing https://nextstrain.org/pathogens page and wouldn't need to be exhaustive. The tiles could also be shrunk further.

<img width="1076" alt="Screenshot 2024-03-08 at 10 29 38 AM" src="https://github.com/nextstrain/nextstrain.org/assets/1176109/f863b0f6-682f-46ea-8be9-3de55c9b6e22">

My thinking here is also influenced by wanting to revamp the "showcase" of dataset quick links on the splash page. We can reuse denser tile design in both places.

Then in the dataset UI make just text suggestions for the filter box.

![filter-examples](https://github.com/nextstrain/nextstrain.org/assets/1176109/b1e1a9a2-bc0e-4635-8f9b-534e0b703c82)

We're not going to have easy tiles for dataset UI for datasets under `/groups` and text could be more flexible in this case.

trvrb commented:
One more small suggestion: Over the past couple days, I was periodically decrufting the `nextstrain-data` bucket of the extraneous `ncov_` datasets. However, I generally couldn't really tell when the indexer ran and whether it made sense to walk through again. I could see it as useful to have a sentence at the very bottom of the UI that reads "Datasets indexed X hours ago." And when you hover you get a UTC timestamp, essentially how commit times work on GitHub with relative being the text on the page and absolute being on hover. Not essential (and could be left off PR), but it would also seem like a particularly useful thing when we extend this UI to `/groups` where people might be confused about a recent upload not showing up.

jameshadfield commented:
This PR is ready for re-review -- I'd like to merge shortly as it only affects the /pathogens and /staging pages and is a big improvement to them (in my opinion). Some points to keep in mind:

* I can't rebuild the review app ([slack thread](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1711422392246149)), so to test this you'll need to run it locally ☹️ 
* Narratives are not yet handled. The previous UI didn't surface them either so this isn't a regression. The SC2 narratives are still available at /sars-cov-2
* There's currently 2 extra zika datasets shown in the UI -- these have been removed from the bucket but will take time to propagate through to the index. I'll cross this point out once it happens.

**Things I'd like to push to subsequent PRs:**

* [@trvrb] Surfacing the time the index was taken. This is eminently doable as the S3 inventory includes a `creationTimestamp` which is [the start time and the date when the inventory report generation begins scanning the bucket](https://github.com/awsdocs/amazon-s3-developer-guide/blob/master/doc_source/storage-inventory.md#what-is-an-inventory-manifest). 
* [@trvrb] Sorting the resources (within each pathogen) by the order they appear in our [core manifest](https://github.com/nextstrain/nextstrain.org/blob/master/data/manifest_core.json). They are currently sorted alphabetically or by last updated (depending on UI configuration).

**responses to (most) review comments above**

> [@huddlej] blocking request would be for the default grouping to be by "pathogen (sorted alphabetically)" instead of the most recently updated

> [@joverlee521] Thoughts on separating grouping and sorting options? I'd like to group by pathogens but sort by published date, where the top entry is the pathogen with the more recent dataset.

> [@trvrb] Keep one pathogen per card and sort cards alphabetically by pathogen name or by last updated date

> [@huddlej] I wish that each card’s header specified the human-readable name of the dataset like “SARS-CoV-2 datasets last updated 2024-02-28” instead of “Datasets last updated 2024-02-28”.... Some of the cards include datasets for multiple distinct pathogens or appear incomplete. For instance, there is a card with a single mpox lineage B.1 dataset followed by another card with two more mpox datasets and the seasonal flu B/Vic dataset (below). Similarly, Zika is grouped with subsets of Vic, H3, and H1 datasets.

The grouping is now always by-pathogen, with two sorting approaches (hover over the sorting options to see a full explanation of how they work). The default sorting is alphabetical. I plan to revisit the within-pathogen sorting when I implement the sorting-according-to-manifest-order (see above). 

> [@huddlej] I wish the icons for “total number of datasets” and “total number of versions” were a little more clearly mapped to those concepts.

> [@joverlee521] Echoing the icons are misleading for datasets/snapshots. I'd suggest a bullet list for datasets and albums for snapshots.

> [@trvrb] Modify the icons for dataset count and snapshot count. 

Icons updated 😄 

> [@joverlee521] The word "updated" can be misleading because here it means when the dataset was uploaded to S3, maybe "published" is more appropriate? Or just rely on the word "snapshots", i.e. "Snapshots between 2023-12-13 & 2024-02-06"

> [@trvrb]  I'd suggest replacing "Last updated between 2020-02-21 & 2024-03-05" with "Last updated 2024-03-05", just to be less verbose.

Updated to just use "Most recent snapshot: ...". Even this is a little misleading because if it's updated in the last few days the index won't know about it. I've tried to add this caveat where I can, but let me know if you can think of a better way to present this.

> [@joverlee521] Do you plan to add links to the individual pathogen pages? (e.g. https://nextstrain.org/sars-cov-2/)

I've added this as a quicklink for the SC2 group after discussion with Trevor. While we have a page for /influenza, it contains no information beyond what I'm presenting here so I've not made a link to it.

> [@trvrb] Collapse carat is hard to restore due to it jumping down the page. Suggest moving the carat to the top of the card.

Switched back to this (it was actually my initial implementation) 

> [@joverlee521] The per dataset sparklines take up a lot of room on the page. Maybe they can be summarized as a single sparkline per pathogen and only display the per dataset sparklines on hover?

> [@huddlej] I wish that when I hovered over the sparkline for each dataset that I saw the same details on demand that I get when I hover over the dataset link.

I've ended up dropping the KDEs / sparklines and gone with a text-based summary. Let me know what you think!

> [@joverlee521] I was initially confused by the empty pipes, I thought there was some UI issue that prevented the words from loading. I wonder if a tree view of the datasets would help here?

I've improved these a bit - hopefully they're not still confusing? I wanted to avoid UIs which required lots of clicking...

> [@trvrb] Quicker main link. Ie clicking on card title of "rsv" or "ncov" should be the equivalent of going to /rsv or /ncov.

Done

> [@trvrb] The modal box with individual snapshots doesn’t flex well between smaller collections and larger collections....

> [@huddlej] For the modal window of past snapshots, I wish that the dots were bigger (easier to hover over and click, although I see now that the size varies by dataset and maybe by number of points in each column) and that there was a y-axis that visually indicated the order of the dots in each column. It also looks like dots aren’t grouped into columns by month since the same column can have snapshots from October, November, and December. What if they were actually grouped by month?

Modal redone and much improved - UI described in [the companion docs update](https://nextstrain--196.org.readthedocs.build/en/196/guides/snapshots.html#viewing-all-available-core-resources-and-their-snapshots)

> [@huddlej] Also, what if the modal window had a standard “x” icon in a top corner ... Also also, you probably don’t have to literally type “fine print” 😆

Done and done!

trvrb commented:
Awesome! I really appreciate attending to all the various issues. Just a couple quick notes:

1. I'm not getting any difference in behavior between the "alphabetical" sort and the "most recently updated" sort. Both seem to be most recently updated with zika currently at the top of the list:

<img width="1112" alt="Screenshot 2024-03-28 at 8 57 05 PM" src="https://github.com/nextstrain/nextstrain.org/assets/1176109/2f09544b-ab9b-42ff-a61e-351d43f9b9b0">

2. Remove the "(monkeypox)" parenthetical. I see how you meant this to be clarifying, but it reads as "monkeypox" is the real name for the pathogen that we short hand of "mpox" for, kind of like how "SARS-CoV-2" is the real name of the pathogen that we short hand as "ncov". However, "monkeypox" is fully deprecated as a name.

<img width="1112" alt="Screenshot 2024-03-28 at 8 58 34 PM" src="https://github.com/nextstrain/nextstrain.org/assets/1176109/cd7474c5-318e-49f7-9b09-adb6b0217409">

3. For "flu (Influenza)". Replace with "flu (seasonal and avian influenza)". It's not capitalized.

joverlee521 commented on static-site/src/components/ListResources/IndividualResource.jsx:
```diff
@@ -0,0 +1,138 @@
+export const IndividualResource = ({data, isMobile}) => {
+  const ref = useRef(null);
+  const [topOfColumn, setTopOfColumn] = useState(false);
+  useEffect(() => {
+    /* The column CSS is great but doesn't allow us to know if an element is at
+    the top of its column, so we resort to JS */
+    if (ref.current.offsetTop===ref.current.parentNode.offsetTop) {
+      setTopOfColumn(true);
+    }
+  }, []);
```
Seems like this does not always get reset when changing the sorting order. 
For pathogens with a lot of datasets, this can mess up the tree listing
 
![Screenshot 2024-04-01 at 5 02 15 PM](https://github.com/nextstrain/nextstrain.org/assets/40774762/b8cd9992-bac6-45e0-be82-a119d2214b10)

jameshadfield replied:
Great observation! This resulted in two changes.

The bug is due to `<IndividualResource>` elements being tracked by React (via their unique `key`) and as we change sorting methods they re-render but aren't re-created and thus `useEffect` doesn't fire. There are a few ways to solve it but I made the key depend on the sorting method, as that's the approach we use to avoid this exact bug for window resizes, collapse state etc.

The second change I discovered while debugging this - I had been using `useEffect` within hooks `useFilterOptions` and `useSortAndFilter`. These effects were running after the components had been rendered (that's how `useEffect` works), which ultimately resulted in every `<IndividualResource>` rendering three times when it should only render once. I switched to `useMemo` which is the appropriate hook and fixes this. (Happy to talk through this more if anyone's interested.)

joverlee521 replied:
>  I made the key depend on the sorting method, as that's the approach we use to avoid this exact bug for window resizes, collapse state etc.

Ah, thanks for explaining, I did not fully understand that updating the key will force a re-render, but that makes sense!

joverlee521 reviewed (commented):
Thanks for all of the improvements! I tried to do a more in-depth code review this time as well. I flagged some questions, but they're mostly small things. 

Really cool to see this come together!

jameshadfield commented on src/endpoints/listResources.js:
```diff
@@ -0,0 +1,28 @@
+};
+
+export const listResources = contentTypesProvided([
+  ["text/html", (req, res, next) => next()], /* go to to next acceptable routing middleware (currently Gatsby) */
```
> Is that the intention here?

Yes, exactly. For those accept types (e.g. browsers) this route effectively doesn't exist. Only if you accept JSON does the API route show up.

jameshadfield replied:
(I played with the idea of adding a JSON-only route for the actual page being rendered, e.g. `/pathogens` and `/staging`, and I still think this is an interesting direction. But I'm not sure it's the best. Either way, the API response clearly indicates it's not stable and we should feel free to change the routes at will.)

joverlee521 replied:
> I played with the idea of adding a JSON-only route for the actual page being rendered, e.g. /pathogens and /staging

Ah, that's an interesting idea! I think it makes sense but definitely can come back to this later.

jameshadfield replied:
Added a link to this discussion [in force push.](https://github.com/nextstrain/nextstrain.org/compare/dd67afcd4196f21f79cbde61fe002d4db70ee738..5f416c204dde13b966df7a37d4dddf4a73bd6954) If pressed I would prefer using the routes in this proposal (/pathogens etc) but something for later discussion and trivial to change the route down the road.

jameshadfield commented on src/routing/listResources.js:
```diff
@@ -0,0 +1,15 @@
+ * warning message included in the response indicating this.
+ */
+export function setup(app) {
+  app.routeAsync("/list-resources/:page")
```
Will do. Note that the API may change in the future - we may want to display pages which accumulate resources across different sources. But for the time being `sourceName` is better than `page`.

jameshadfield replied:
Updated route param and added a comment in the code similar to my one in this thread

jameshadfield commented on static-site/src/components/ListResources/index.jsx:
```diff
@@ -31,6 +32,8 @@ function ListResources({sourceId, versioned=true, elWidth}) {
   useSortAndFilter(sortMethod, selectedFilterOptions, originalData, setResourceGroups)
   const availableFilterOptions = useFilterOptions(resourceGroups);
+  const [modal, setModal ] = useState(null);
+  const dismissModal = useCallback(() => setModal(null), [])
```
Removed (see above)

jameshadfield commented on static-site/src/components/ListResources/useSortAndFilter.js:
```diff
@@ -0,0 +1,46 @@
+        break;
+      case "alphabetical":
+        _sortGroups = (groupA, groupB) => groupA.groupName.toLowerCase() > groupB.groupName.toLowerCase();
+        _sortResources = (a, b) => a.name < b.name
```
It's not a typo, I just habitually use boolean sorting 🤦 (I must remember to stop doing this in JS!)

I've re-written the sort functions (they were all wrong!) which should also resolve [the observation Trevor made](https://github.com/nextstrain/nextstrain.org/pull/803#issuecomment-2026617139)

jameshadfield commented on static-site/src/components/ListResources/index.jsx:
```diff
@@ -0,0 +1,165 @@
+        '<br/>' +
+        'Datasets within each group ordered by their latest update date.'}>
+        most recently updated
+      </TooltipWrapper>
```
Done - thanks!

joverlee521 replied:
Ah, the `input` and `label` need to switch `id` and `htmlfor`

```diff
-      <input htmlFor="alphabetical" type="radio" onChange={onChangeValue} value="alphabetical" checked={"alphabetical"===sortMethod} />
+      <input id="alphabetical" type="radio" onChange={onChangeValue} value="alphabetical" checked={"alphabetical"===sortMethod} />
       <TooltipWrapper description={'Pathogen groups ordered alphabetically. ' +
         '<br/>' +
         'Datasets within each group ordered alphabetically.'}>
-        <label id='alphabetical'>alphabetical</label>
+        <label htmlfor='alphabetical'>alphabetical</label>
       </TooltipWrapper>
-      <input htmlFor='lastUpdated' type="radio" onChange={onChangeValue} value="lastUpdated" checked={"lastUpdated"===sortMethod}/>
+      <input id='lastUpdated' type="radio" onChange={onChangeValue} value="lastUpdated" checked={"lastUpdated"===sortMethod}/>
       <TooltipWrapper description={'Pathogen groups ordered by the most recently updated dataset within that group. ' +
         '<br/>' +
         'Datasets within each group ordered by their latest update date.'}>
-        <label id='lastUpdated'>most recently updated</label>
+        <label htmlfor='lastUpdated'>most recently updated</label>
```

jameshadfield replied:
🤦  - how do you I these to be sure I've got it right?

joverlee521 replied:
I tested by clicking on the text of the radio buttons. The current testing app only changes the selected radio button when I click on the radio button itself. 

https://github.com/nextstrain/nextstrain.org/assets/40774762/ad0375de-11d3-49df-bd6f-b3557af78f9c

jameshadfield replied:
Oh, I totally misunderstood the `<label>` element, I thought it was for screen readers. This is a nice improvement. I've also changed the cursors to better indicate their clickability based on [a little reading](https://kizu.dev/cursor-pointer/). [Changed in force push.](https://github.com/nextstrain/nextstrain.org/compare/dd67afcd4196f21f79cbde61fe002d4db70ee738..5f416c204dde13b966df7a37d4dddf4a73bd6954)

jameshadfield commented on static-site/src/components/ListResources/IndividualResource.jsx:
```diff
@@ -0,0 +1,138 @@
+  return (
+    <div>
+      <div onMouseOver={onMouseOver} onMouseOut={onMouseOut}>
+        {React.cloneElement(children, { hovered })}
```
😄

jameshadfield commented on static-site/src/components/ListResources/IndividualResource.jsx:
```diff
@@ -0,0 +1,138 @@
+  const onMouseOut = useCallback(() => setHovered(false), [setHovered]);
+  return (
+    <div>
+      <div onMouseOver={onMouseOver} onMouseOut={onMouseOut}>
```
Yup, changed - I got in the habit of using `useCallback` because it makes it straightforward to memoise components or diff props. (If you don't then the function gets recreated each render and so it's always different.) But we don't use those performance techniques here so no need to use it!

jameshadfield commented:
Thanks for the review @joverlee521 and comments @trvrb! All fixed, and I plan to merge in tomorrow if there are no further comments.

trvrb commented:
Awesome! This is really great work @jameshadfield. Will be immediately useful at `/pathogens` and will be a good basis for other pages as well. The updated modal with the bubbles and the timeline is especially clever. It blends playful and useful nicely. Please merge whenever you'd like.

jameshadfield commented:
Force pushed to fix up [the final review comment](https://github.com/nextstrain/nextstrain.org/pull/803#discussion_r1550189398). [CI failed](https://github.com/nextstrain/nextstrain.org/actions/runs/8545926818/job/23415271637) due to a spurious / unrelated / stochastic failure:
```
getAvailable community URLs correctly interpret the GitHub branch › Explicit non-default branch
```
I [manually retriggered CI](https://github.com/nextstrain/nextstrain.org/actions/runs/8545977500) which passed, so I'm merging this now.

(As an aside, I'm noticing more and more stochastic CI failures. Or maybe I've just been doing more work on nextstrain.org lately and they've always been here.)
